packages/material-ui-lab/src/TimelineOppositeContent/TimelineOppositeContent.js

```diff
@@ -1,41 +1,77 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
+import { deepmerge } from '@material-ui/utils';
 import { capitalize } from '@material-ui/core/utils';
-import { withStyles } from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import Typography from '@material-ui/core/Typography';
 import TimelineContext from '../Timeline/TimelineContext';
 import TimelineItemContext from '../TimelineItem/TimelineItemContext';
+import { getTimelineOppositeContentUtilityClass } from './timelineOppositeContentClasses';
 
-export const styles = () => ({
-  /* Styles applied to the root element. */
-  root: {
-    padding: '6px 16px',
-    marginRight: 'auto',
-    textAlign: 'right',
-    flex: 1,
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+  return deepmerge(
+    {
+      ...styles[`align${capitalize(styleProps.align)}`],
+    },
+    styles.root || {},
+  );
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { align, classes } = styleProps;
+
+  const slots = {
+    root: ['root', `align${capitalize(align)}`],
+  };
+
+  return composeClasses(slots, getTimelineOppositeContentUtilityClass, classes);
+};
+
+const TimelineOppositeContentRoot = experimentalStyled(
+  Typography,
+  {},
+  {
+    name: 'MuiTimelineOppositeContent',
+    slot: 'Root',
+    overridesResolver,
   },
+)(({ styleProps }) => ({
+  /* Styles applied to the root element. */
+  padding: '6px 16px',
+  marginRight: 'auto',
+  textAlign: 'right',
+  flex: 1,
   /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
+  ...(styleProps.align === 'right' && {
     textAlign: 'left',
-  },
-});
+  }),
+}));
 
-const TimelineOppositeContent = React.forwardRef(function TimelineOppositeContent(props, ref) {
-  const { classes, className, ...other } = props;
+const TimelineOppositeContent = React.forwardRef(function TimelineOppositeContent(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTimelineOppositeContent' });
+  const { className, ...other } = props;
 
   const { align = 'left' } = React.useContext(TimelineContext);
   const { classes: contextClasses = {} } = React.useContext(TimelineItemContext);
```

Comment:
We won't be needing this after https://github.com/mui-org/material-ui/pull/25822#pullrequestreview-638692509 No preference on the other of how there are merged as long as we release them all at once.

Reply:
Let's merge this one and we can actually do the update in https://github.com/mui-org/material-ui/pull/25822#pullrequestreview-638692509